flow/Net2.actor.cpp

```diff
@@ -620,6 +681,10 @@ class SSLConnection : public IConnection, ReferenceCounted<SSLConnection> {
 			self->closeSocket();
 			connected.sendError(connection_failed());
 		}
+
+		if(usePool) {
+			--N2::g_net2->sslPoolHandshakesInProgress;
```

Comment:
nit:  This would be easier to read if it were on the line after `wait( onHandshook )`.  I'm guessing it's here for exception handling.  Add a comment, or (if you have free time) use RAII, and replace `usePool` with something that decrements the count on deallocation.

Reply:
Agreed.  Do you know of a generic class for this kind of thing?  I have one in Redwood but it's not committed yet and I wanted to make it a template class and play with its interface a bit to make it more convenient.

Reply:
This is done, via a new utility class called Hold<T>.  I'm not a huge fan of the name but it's the best I could come up with.